java/arcs/core/storage/DirectStoreMuxer.kt

```diff
@@ -52,9 +52,8 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
         if (!storeRecord.store.closed) {
             log.debug { "close the store(${storeRecord.id})" }
 
-            // The store will be actually closed as soon as there are no connected proxies.
```

Comment:
Just wanted to understand -- this comment reads as if we are intentionally not closing, presumably due to performance benefits. By explicitly closing, are we trading off removal of risk of leaks for more cost somewhere else such as more frequent need to read a store's value again?

Reply:
Yep, a backing store gets restored from underlying storage upon subsequent accesses if it was already closed.

It is closed when 
a) it is evicted from lru cache map (so unbounded to bounded)
b) all clients/callbacks at its hosting reference mode store are deregistered i.e. all handles closed (so bounded to 0)

Lru map size for each backing store (storage key) is 50 with current configuration.

Reply:
I think this change is good. But maybe check with Jason. I think the leak was introduced with https://github.com/PolymerLabs/arcs/pull/5588/files